napari/_qt/qt_event_loop.py

```diff
@@ -1,29 +1,80 @@
+import os
 import sys
 from contextlib import contextmanager
-from os.path import dirname, join
 
 from qtpy.QtCore import Qt
-from qtpy.QtGui import QPixmap
+from qtpy.QtGui import QIcon, QPixmap
 from qtpy.QtWidgets import QApplication, QSplashScreen
 
+from napari import __version__
+
 from ..utils.perf import perf_config
 from .exceptions import ExceptionHandler
 from .qthreading import wait_for_workers_to_quit
 
+NAPARI_ICON_PATH = os.path.join(
+    os.path.dirname(__file__), '..', 'resources', 'logo.png'
+)
+NAPARI_APP_ID = f'napari.napari.viewer.{__version__}'
+
+
+def set_app_id(app_id):
+    if os.name == "nt" and app_id and not getattr(sys, 'frozen', False):
+        import ctypes
+
+        ctypes.windll.shell32.SetCurrentProcessExplicitAppUserModelID(app_id)
+
+
+def get_app(
+    app_name: str = 'napari',
+    app_version: str = __version__,
+    icon: str = NAPARI_ICON_PATH,
+    org_name: str = 'napari',
+    org_domain: str = 'napari.org',
+    app_id: str = NAPARI_APP_ID,
+) -> QApplication:
+    """Get or create the Qt QApplication.
 
-def get_app() -> QApplication:
-    """Get or create the Qt QApplication
+    There is only one global QApplication instance, which can be retrieved by
+    calling get_app again, (or by using QApplication.instance())
+
+    Parameters
+    ----------
+    app_name : str, optional
+        Set app name (if creating for the first time), by default 'napari'
```

Comment:
If the app already exists has the name `napari`, and then you call it with a setting here like `my_custom_name`, will it get renamed as such? If so then I might change `if creating for the first time`, if not what are your thoughts about emitting a warning if the user tries to set it and they are unable to?

This applies to all of these similar input args

Reply:
nothing gets changed on the second call to get_app.  I added a warning. I also made it so that those kwargs are "all or nothing".  (i.e. if none of them are set, then it uses the napari defaults, if any are provided, it only uses the provided values)

Reply:
Great, thanks!